.github/workflows/external-application-connection-tests.yml

```diff
@@ -31,7 +31,147 @@ jobs:
+          openssl pkcs12 -in ssl.p12 -nocerts -noenc -out sslserver.key -password pass:myPassword
+          
+      - name: Create postgresql Docker file
+        run: |
+          cat > Dockerfile-Postgresql <<EOF
+          FROM postgres AS postgres-ssl
+
+          # Copy certificates
+          COPY sslserver.key /var/lib/postgresql/server.key
+          COPY sslserver.crt /var/lib/postgresql/server.crt
+          RUN chown postgres:postgres /var/lib/postgresql/server.crt && \
+              chown postgres:postgres /var/lib/postgresql/server.key && \
+              chmod 600 /var/lib/postgresql/server.key
+          EOF
+
+      - name: Build postgrsql image with certificates
+        uses: docker/build-push-action@v3
+        with:
+          context: .
+          tags: postgres-ssl
+          target: postgres-ssl
+          file: Dockerfile-Postgresql 
+
+      - name: Deploy postgresql
+        run: |
+          docker run -d --name postgresql \
+              --hostname postgresql.example.com \
+              -e POSTGRES_PASSWORD=mysecretpassword \
+              -e POSTGRES_USER=jss \
+              postgres-ssl -c ssl=on \
+              -c ssl_cert_file=/var/lib/postgresql/server.crt \
+              -c ssl_key_file=/var/lib/postgresql/server.key
+
+      - name: Connect DB container to network
+        run: docker network connect example postgresql --alias postgresql.example.com
+
+      - name: Build the tests
+        run: docker exec jss ./build.sh --work-dir=./build
+
+      - name: Test connection with java
+        run: |
+          docker exec jss mkdir /root/.postgresql
+          docker exec jss cp /root/sslserver.crt /root/.postgresql/root.crt
+          docker exec -t jss sh -c 'java -cp \
+               "/usr/share/java/*:/usr/share/java/ongres-stringprep/*:/usr/share/java/ongres-scram/*:usr/lib/java/jss.jar:/usr/share/java/slf4j/*:/root/jss/build/classes/tests" \
+              org.mozilla.jss.tests.JSSConnectionPostgres jss mysecretpassword \
+              '"'"'jdbc:postgresql://postgresql.example.com:5432/jss?ssl=true&sslmode=verify-full'"'" | tee output-java
+          grep "Connection DONE" output-java
+
+      - name: Create JSS DB and configuration files
+        run: |
+          cat > java.security <<EOF
+          security.provider.1=org.mozilla.jss.JSSProvider /root/jss/jss.cfg
+          security.provider.2=sun.security.provider.Sun
+          security.provider.3=sun.security.ssl.SunJSSE
+          security.provider.4=sun.security.rsa.SunRsaSign
+          security.provider.5=sun.security.ec.SunEC
+          security.provider.6=com.sun.net.ssl.internal.ssl.Provider
+          security.provider.7=com.sun.crypto.provider.SunJCE
+          security.provider.8=sun.security.jgss.SunProvider
+          security.provider.9=com.sun.security.sasl.Provider
+          security.provider.10=org.jcp.xml.dsig.internal.dom.XMLDSigRI
+          security.provider.11=sun.security.smartcardio.SunPCSC
+          EOF
```

Comment:
Instead of creating a `java.security` file (which affects the whole JVM) we probably can use the `sslfactory` param with a custom `SSLSocketFactory` just for the PostgreSQL connection:
https://jdbc.postgresql.org/documentation/use/#connection-parameters
This way it might be easier to use and would avoid conflicts with everything else running in the JVM.

Reply:
According to the documentation the `sslfactory` is string with the name of the class to instantiate as socket factory. Additionally:

> This class must have a zero argument constructor or a single argument constructor 

In **JSS** the only  factory I have seen is `org.mozilla.jss.ssl.javax.JSSSocketFactory` and this has a constructor with multiple parameters so cannot be directly used. If a new factory has to be implemented then I do not think make sense for this simple test but maybe I am missing something. 
@edewata, is there another approach without adding a new security provider, via `java.security` or API?

Reply:
Yeah, we can't do this with the existing code, but it's some to think about for future improvements, and hopefully it won't require major changes. To my understanding this is a pretty common way to provide a custom socket, so we should try to support this if possible.

Reply:
Yes, I do not think this is a peculiarity of _Postgresql_ so in the future we should work on that. Do we should create an issue for that as a remind?

I am going to merge for now.

For the _ACME_ based test, I move it to __pki__ or just drop it?

Reply:
Yes, please create a ticket for the socket factory and move the ACME test to PKI. That would be a good addition to PKI CI. Thanks!